jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/ArrayByteBufferPool.java

```diff
@@ -444,9 +445,9 @@ private Pool.Entry<RetainableByteBuffer> findOldestEntry(long now, Pool<Retainab
         private final Pool<RetainableByteBuffer> _pool;
         private final int _capacity;
 
-        private RetainedBucket(int capacity, int size)
+        private RetainedBucket(int capacity, int poolSize)
         {
-            _pool = new ConcurrentPool<>(ConcurrentPool.StrategyType.THREAD_ID, size, true);
+            _pool = new CompoundPool<>(ConcurrentPool.StrategyType.THREAD_ID, poolSize, true);
```

Comment:
I would say no. We've already noticed that `ArrayByteBufferPool` exposes too many details of the old implementation as constructor arguments and I don't want to corner us in case we want to change the implementation yet again in the future.

There's always the option of using a different `ByteBufferPool` implementation in the worst case.